src/test/java/no/unit/nva/model/PublicationTest.java

```diff
@@ -31,4 +43,49 @@ protected Object produceFramedPublication(JsonNode publicationWithContext) throw
         options.setPruneBlankNodeIdentifiers(true);
         return JsonLdProcessor.frame(input, frame, options);
     }
+
+    @Test
+    public void updatingDoiRequestStatusForValidTransitionChangeReturnsUpdatedDoiRequestStatus()
+        throws InvalidIssnException, MalformedContributorException {
+        var publication = PublicationGenerator.generateJournalArticlePublication();
+        publication.updateDoiRequestStatus(REJECTED);
+        assertThat(publication.getDoiRequest().getStatus(), is(equalTo(REJECTED)));
+
+        publication.updateDoiRequestStatus(APPROVED);
+        assertThat(publication.getDoiRequest().getStatus(), is(equalTo(APPROVED)));
+    }
+
+    @Test
+    public void updatingDoiRequestStatusWithInvalidTransitionRequestThenThrowsIllegalArgumentException()
+        throws InvalidIssnException, MalformedContributorException {
+        var publication = generatePublicationWithRejectedDoiRequestStatus();
+
+        var actualException = assertThrows(IllegalArgumentException.class,
+            () -> publication.updateDoiRequestStatus(REQUESTED));
+        assertThat(actualException.getMessage(),
+            is(equalTo(String.format(ERROR_MESSAGE_NOT_ALLOWED_TO_CHANGE_STATUS_FROM_S_TO_S, REJECTED, REQUESTED))));
```

Comment:
use public static method for the error message

Reply:
What are you trying to assert here? That the message is the same as the one in the Publication class? that the message contains both REJECTED and REQUESTED statuses? That the message carries some kind of information? 

If we change the message and we omit for example the new status, then the person who will make the change will compare the line you have written with the line in the code and they think, they are the same so I change the test here and I am OK. But if you explicitly test for the presence of the two statuses then you have made your expectation clear. 

My point is that my understanding of this line is that you expect the message specified in the Publication class to appear to the test. So it is not directly linked to the class but it is indirectly because copying the returned value means 'I expect the specific value'

Reply:
How about testing if bot REJECTED and REQUESTED statuses are contained in the message?